### PR TITLE
[WUD-126] Myeatman/manipulation api

### PR DESCRIPTION
**Pull request by myeatman-bdai**
This adds a stub for adding support for the manipulation api. Tested on robot.

### CODE REVIEW

**Review by jbarry-bdai — COMMENTED** [review]
Looks like the right idea!

You can edit the robot command handling so that there's not so much copy paste code - I'm pretty sure we're currently the only people using that.

File: spot_driver/spot_driver/spot_ros2.py [hunk 1]
```diff
@@ -1191,6 +1253,10 @@ def main(args=None):
                                                                spot_ros.handle_robot_command,
                                                                callback_group=spot_ros.group)
 
+        spot_ros.manipulation_server = ActionServer(node, Manipulation, 'manipulation',
+                                                              spot_ros.handle_manipulation_command,
+                                                              callback_group=spot_ros.group)
```
Comment (amessing-bdai) [inline spot_driver/spot_driver/spot_ros2.py:1258]: Should this be conditional on the robot having an arm?

**Review by jbarry-bdai — COMMENTED** [review]
This looks right but I think there's a lot of copy/pasted code from robot command.  Could we create one function that does the bulk of the work?

**Comment by myeatman-bdai** [comment]
> This looks right but I think there's a lot of copy/pasted code from robot command. Could we create one function that does the bulk of the work?

I believe it's possible to create a function that shares some of the workload, but if you do the diff between the whole `handle_robot_command` function and `handle_manipulation_command` I believe there's less similarity than you're hoping for. If we had a 3 or more duplicated workflows, like an additional `handle_flying_command`, I'd be much more onboard. But I'd like to just move on to actually trying to grasp something in the diagnose-repair workflow.

**Review by jbarry-bdai — APPROVED** [review]

**Pull request merged**